### PR TITLE
SITES-450 demo script

### PR DESCRIPTION
**Pull request by freshtonic**

### CODE REVIEW

**Comment by tessereth** [comment]
please update changelog post 0.8.1 release

File: spec/database_dump_loader.rb [hunk 1]
```diff
@@ -0,0 +1,52 @@
+
+# Loads a Postgres database dump.
+#
+# Only COPY blocks are extracted. Any other commands, comments, whitespace is ignored.
+#
+# Skips content_blocks (because it's still in prod, and schema_migrations
+# (because db:test:prepare will already have popuated it and you'll get unique
```
Comment (coderdan) [inline spec/database_dump_loader.rb]: Minor typo

File: spec/database_dump_loader.rb [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+# key violations).
+#
+class DatabaseDumpLoader
```
Comment (coderdan) [inline spec/database_dump_loader.rb]: This is nice!

File: spec/database_dump_loader.rb (hunk 1 above)
Comment (freshtonic) [inline spec/database_dump_loader.rb]: ah yes

File: spec/features/demo_spec.rb [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+    expect(page.current_path).to eql("/")
+    expect(page.title).to eql("GovAuBeta")
+    expect(page).to have_css('h2', :text => 'Information and services')
+    expect(page).to have_css('h2', :text => 'About Australia')
+    visit(page.find('.next-holiday a')[:href])
+
+    expect(page.current_path).to eql("/times-and-dates/australian-public-holidays")
+    expect(page).to have_css('h1', :text => 'Australian Public holidays')
+    visit(page.find('a', :text => 'School holidays and term dates')[:href])
```
Comment (coderdan) [inline spec/features/demo_spec.rb]: Is there a reason you're doing this instead of click? Does it even matter?

**Comment by coderdan** [comment]
One small query but otherwise LGTM

File: spec/features/demo_spec.rb [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+  end
+
+  it "should be possible to demo the application" do
+    visit '/'
+
+    expect(page.current_path).to eql("/")
+    expect(page.title).to eql("GovAuBeta")
+    expect(page).to have_css('h2', :text => 'Information and services')
+    expect(page).to have_css('h2', :text => 'About Australia')
+    visit(page.find('.next-holiday a')[:href])
```
Comment (micapam) [inline spec/features/demo_spec.rb]: This isn't a problem with the test per se, but it exposes an accessibility problem... should be able to use something other than CSS class.

File: spec/features/demo_spec.rb (hunk 3 above)
Comment (freshtonic) [inline spec/features/demo_spec.rb]: @coderdan does `click_link` take a selector? There is one case in the demo where there are two links with same text on the page.

Comment (coderdan) [inline spec/features/demo_spec.rb]: Not sure. Looking it up now...

File: spec/features/demo_spec.rb [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+    visit(page.find('article.content-listing li a', :text => 'Department of Communications and the Arts')[:href])
+
+    expect(page.current_path).to eql('/department-of-communications-and-the-arts')
+    visit(page.find('aside li a', :text => 'Our role')[:href])
+
+    expect(page.current_path).to eql('/department-of-communications-and-the-arts/our-role')
+    visit(page.find('aside li a', :text => 'Careers')[:href])
+
+    expect(page.current_path).to eql('/department-of-communications-and-the-arts/careers')
+    visit(page.find('aside li a', :text => 'Our role')[:href])
```
Comment (micapam) [inline spec/features/demo_spec.rb]: I think these precise selectors make the test too closely tied to the implementation = too fragile. Should only go to that level of detail if there's actual ambiguity on the page (e.g. more than one 'Our role' link).

Comment (freshtonic) [inline spec/features/demo_spec.rb]: There is a case in the demo where two links with the same text appear on the page (dept of comms is in the footer can can appear in main content, so need to disambiguate).

File: spec/features/demo_spec.rb (hunk 3 above)
Comment (coderdan) [inline spec/features/demo_spec.rb]: You might able to do something like:

    page.find('a', :text => 'School holidays and term dates').click

See http://www.rubydoc.info/github/jnicklas/capybara/master/Capybara/Node/Element#click-instance_method

File: spec/features/demo_spec.rb (hunk 5 above)
Comment (freshtonic) [inline spec/features/demo_spec.rb]: I think `aside li a` can be `aside a`. The demo google doc *specifically* says "click a link in the sidebar", so I think that's reasonable.

Comment (micapam) [inline spec/features/demo_spec.rb]: Yeah - that sounds like a good balance between disambiguation & the semantics > syntax principle.

**Comment by freshtonic** [comment]
@coderdan so regarding `click_link` versus `visit(page.find('aside li a', :text => 'Our role')[:href])`

**Comment by coderdan** [comment]
Honestly @freshtonic I don't think its an issue. I was just curious to know the motivation. As I said: LGTM :)

**Comment by micapam** [comment]
LGTM !

**Pull request merged**